app/events/telegram.go

```diff
@@ -276,6 +272,34 @@ func (l *TelegramListener) sendBotResponse(resp bot.Response, chatID int64) erro
 	return nil
 }
 
+// sendMdWithFallback sends message with markdown mode and fallback to plain text
+func (l *TelegramListener) sendMdWithFallback(resp bot.Response, chatID int64) (tbapi.Message, error) {
+	log.Printf("[DEBUG] sending message to telegram")
+	tbMsg := tbapi.NewMessage(chatID, resp.Text)
+	tbMsg.ParseMode = tbapi.ModeMarkdown
+	if resp.ParseMode != "" {
+		tbMsg.ParseMode = resp.ParseMode
```

Comment:
The unfilled `ParseMode` switches to [legacy markdown](https://core.telegram.org/bots/api#markdown-style). Please see [documentation](https://core.telegram.org/bots/api#formatting-options) for details. So, the problem you are trying to solve will still need to be solved. In that case, you should apply [EscapeTelegramText](https://github.com/go-pkgz/notify/blob/master/telegram.go#L150C6-L156).

Another solution would be escaping a Telegram message according to prohibited symbols in the documentation, but there is no ready-made function for that, and if you add it, it's better to add it to the go-pkgz/notify telegram.go file.

Reply:
I have not found a 100% working way to escape. I used the same function, even extended with more use cases, and still get invalid markdown format errors of all kinds, from time to time. I believe the proposed solution is supposed to do a fallback to plain text. I didn't read the docs, but from my experience, setting ParseMode to an empty string made Telegram ignore the markdown. I'm not sure if this is plain text mode or some limited markdown, but the result looks like plain text to me. This is one of the examples of a message that failed after the escape and was sent with an empty parse mode.

<img width="607" alt="image" src="https://github.com/radio-t/super-bot/assets/535880/3fe2c2ff-6462-4d31-863b-a1c015f1f730">

similar one with a valid tg markdown looks like this

<img width="639" alt="image" src="https://github.com/radio-t/super-bot/assets/535880/cebde39d-5f23-4d61-8f9d-934ed756630e">

Reply:
Yes, I also did some testing in Telegram chat, it also looked like plain text, I didn't notice any formatting.

Reply:
I think you two are right, and I am wrong. It's not written explicitly but it seems like the default way is indeed undocumented PlainText and not legacy Markdown.